cmd/publishing-bot/github.go

```diff
@@ -109,49 +110,26 @@ func CloseIssue(token, org, repo string, issue int) error {
 	return nil
 }
 
-// tail returns lines with a maximum of the given bytes in length. Only complete lines are
-// returned, with the exception if the is only one line, then "..." are put in front.
-// maxBytes must be at least 3.
-func tail(msg string, maxBytes int) string {
-	msg = strings.Trim(msg, "\n")
-	if len(msg) <= maxBytes {
-		return msg
-	}
-	lines := strings.Split(msg, "\n")
-	if len(lines) == 1 {
-		if len(lines[0]) <= maxBytes {
-			return lines[0]
-		}
-		return "..." + lines[0][max(0, len(lines[0])-maxBytes+3):]
-	}
-
-	prefix := "..."
-	ret := []string{}
-	n := len(prefix)
-	for i := len(lines) - 1; i >= 0; i-- {
-		if n+len("\n")+len(lines[i]) > maxBytes {
-			if len(ret) == 0 {
-				return "..." + lines[0][max(0, len(lines[0])-maxBytes+3):]
+func transfromLogToGithubFormat(original string, maxLines int, headings ...string) string {
+	logCount := 0
+	transformed := NewLogBuilderWithMaxBytes(65000, original).
+		AddHeading(headings...).
+		AddHeading("```").
+		Trim("\n").
+		Split("\n").
+		Reverse().
+		Filter(func(line string) bool {
```

Comment:
is this working backwards, i.e. we get the last lines?

Reply:
yeah. it's a little tricky here. i did a reverse before filtering and reverse again to restore the order.

Reply:
plz see the test i wrote for it